kagami/routes/deps.py

```diff
@@ -1,8 +1,12 @@
 from typing import Annotated
 
 from fastapi import Depends
+from sqlalchemy.ext.asyncio import AsyncSession
 
 from ..core import Supervisor
+from ..database.database_service.engine import DatabaseEngine
 from ..server import supervisor
 
 SupervisorDeps = Annotated[Supervisor, Depends(lambda: supervisor)]
+
+SessionDeps = Annotated[AsyncSession, Depends(lambda: DatabaseEngine.session_factory)]
```

Comment:
DatabaseEngine.session_factory is a function, don't use lambda here.

Reply:
Better to use asynccontextmanager to manage the life cycle of session. If directly use session_factory here, every new request will create a new session and maybe this session will not quit correctly.

Reply:
This was solved by [0eb5c97](https://github.com/ProjectSoftoid/Kagami/pull/12/commits/0eb5c971118ab57e07b650e6f201996e1d786e69)